src/sentry/static/sentry/app/views/groupTags.jsx

```diff
@@ -8,32 +9,51 @@ import LoadingError from '../components/loadingError';
 import LoadingIndicator from '../components/loadingIndicator';
 import {percent, deviceNameMapper} from '../utils';
 import {t} from '../locale';
+import withEnvironment from '../utils/withEnvironment';
 
 const GroupTags = createReactClass({
   displayName: 'GroupTags',
+
+  propTypes: {
+    environment: PropTypes.object,
+  },
+
   mixins: [ApiMixin, GroupState],
 
   getInitialState() {
     return {
       tagList: null,
       loading: true,
       error: false,
+      environment: this.props.environment,
     };
   },
 
   componentWillMount() {
     this.fetchData();
   },
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.environment !== this.props.environment) {
```

Comment:
is it possible not to put this on state at all? and just use it from props?

Reply:
it's certainly possible, but imo this way is just simple since props won't be updated when fetchData is called

Reply:
I don't follow, could you explain later?

Reply:
oh - what about passing it into fetchData as an argument? I just don't like having both props and state but if its not a blocker